### DISCUSSION TITLE
Dinamically disabled page action can't be enabled

### DISCUSSION THREAD

**Discussion by underdpt** [filamentphp/filament#6223] [Help]
Hello,

I'm in a situation where I have the next action:

```php
Actions\Action::make('partialReceive')
    ->icon('fas-warehouse-full')
    ->disabled(function () {
        return $this->unitsToReceive->isEmpty();
    })
    ->form(fn() => $this->getFormReceive())
    ->action(fn(array $data) => $this->receiveOrder($this->record, $data, false)),
````
This is attached to an edit page with a relation manager, where you can select the lines to receive. An event is emmited when you select any line, so `unitsToReceive` is updated.

The isue is that although `unitsToReceive` is not empy, and the closure on  `disable()` gets called, the button is shown disabled.

This is on latest version v2.17.25.

Do you know if there's a way to re-enable an input action this way?

**Comment by danharrin** [comment]
So are you suggesting that the `disabled()` function is being run before the event listener is being run?

**Comment by underdpt** [comment]
I'm not sure which event listener are you referring to. I have a event that's fired on the relation manager, and the listener is run before the `disabled` call (checked with ray). It seems that `disabled()` is checked twice, but on those two checks it returns `false` so is should become enabled:

![imagen](https://user-images.githubusercontent.com/8122137/232289211-b8cbf5f4-5905-4193-b89f-b7772ab03674.png)

**Reply by danharrin** [reply]
I'm not sure then, if its being run in the correct order then it should not be disabled.

**Comment by underdpt** [comment]
Yeah. I've been trying to debug this. I've arrived at `resources/views/components/button.blade.php` where it should render the `disabled` word on the button, around line 84.

When I update any line, the button code is executed, but the issue is that's it's not being rendered: on the Livewire response the button is the original one, not the updated one. I've tested this also by dinamically rendering some string before the button on `resources/views/components/button.blade.php` and those strings aren't on the Livewire response.
